doc/BoxPlatform.md

```diff
@@ -1,71 +1,133 @@
+{
+    // NOTE: This code is specific to your app; it is responsible for retrieving new tokens from a secure server
+    NSLog(@"Attempting to retrieve new access token from server for uniqueId: %@", uniqueId);
 
-@interface AppDelegate ()
+    // Make a request to the secure sever for a token
+    NSString *fullUrl = [NSString stringWithFormat:@"https://YOUR_SECURE_SERVER.example.com/token?id=%@", uniqueId];
+    NSURL *url = [NSURL URLWithString:fullUrl];
 
-@property (nonatomic, readonly, strong) BOXContentClient *boxClient;
+    __block NSString *token;
+    NSURLSessionDataTask *task = [[NSURLSession sharedSession] dataTaskWithURL:url
+            completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                if (error) {
+                    NSLog(@"Failed to retrieve new access token from server");
+                    completion(nil, nil, error);
+                } else {
+                    token = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+                    NSLog(@"Succeeded in retrieving new access token: %@", token);
 
-@end
+                    // Pass token and expiration time back to the client
+                    completion(token, [NSDate dateWithTimeIntervalSinceNow:2700], nil);
+                }
+            }];
+    [task resume];
+};
 
-@implementation AppDelegate
+// Create a client for the authenticated user; the client is keyed to that user's unique ID for caching purposes
+// The unique ID can be any identifier for the user, or a dummy value
+ServerAuthUser *user = [[ServerAuthUser alloc] initWithUniqueID:@"UNIQUE_ID"];
+BOXContentClient *boxClient = [BOXContentClient clientForServerAuthUser:user
+                                                initialToken:nil
+                                                fetchTokenBlockInfo:nil
+                                                fetchTokenBlock:fetchTokens];
 
-- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+// Example of making an API call
+BOXUserRequest *request = [boxClient currentUserRequest];
+[request performRequestWithCompletion:^(BOXUser *user, NSError *error) {
+    if (error) {
+        NSLog(@"Error getting user info");
+    } else {
+        NSLog(@"Retrieved user with login: %@", user.login);
+    }
+}];
+```
 
-    _boxClient = [BOXContentClient clientForServerAuthUniqueId:@"A_UNIQUE_ID"
-                                                  initialToken:nil
-                                           fetchTokenBlockInfo:nil
-                                               fetchTokenBlock:fetchTokenBlock];                                                          
+__Swift:__
+```swift
+let fetchToken: ServerAuthFetchTokenBlock = { (
+        uniqueID: String?,
+        fetchTokenBlockInfo: [AnyHashable: Any]?,
+        completion: ((String?, Date?, Error?) -> Void)!
```

Comment:
Same here. I would think this shouldn't be an IUO, but again, it might be, if we didn't specify nullability.